blocksync/pool.go

```diff
@@ -612,18 +613,16 @@ OUTER_LOOP:
 		bpr.peerID = peer.id
 		bpr.mtx.Unlock()
 
+		to := time.NewTimer(requestRetrySeconds * time.Second)
 		// Send request and wait.
-		bpr.pool.sendRequest(bpr.height, peer.id)
 	WAIT_LOOP:
 		for {
+			bpr.pool.sendRequest(bpr.height, peer.id)
 			select {
-			case <-bpr.pool.Quit():
-				if err := bpr.Stop(); err != nil {
-					bpr.Logger.Error("Error stopped requester", "err", err)
-				}
-				return
-			case <-bpr.Quit():
-				return
+			case <-to.C:
+				bpr.pool.Logger.Debug("Retrying block pool request after timeout", "height", bpr.height, "peer", bpr.peerID)
+				to.Reset(requestRetrySeconds * time.Second)
+				continue WAIT_LOOP
```

Comment:
Would it not be better if we were to retry with a different peer?

Reply:
Yeah, that matches my comment on "turning this case into a block validation failure".

I'll leave it as is for the moment, tho